Lib/test/test_subprocess.py

```diff
@@ -1733,6 +1733,20 @@ def test_run_with_shell_timeout_and_capture_output(self):
                         msg="TimeoutExpired was delayed! Bad traceback:\n```\n"
                         f"{stacks}```")
 
+    def test_encoding_warning(self):
+        code = textwrap.dedent("""\
+            from subprocess import *
+            args = ["echo", "hello"]
```

Comment:
Notice how the majority of the tests in this file use a child process of `[sys.executable, "-c",  "print('hello')"]` to avoid the platform problems. Depending on a shell command working across platforms always causes future problems. See also the above test which simply skips windows because it relies on sleep 3 instead of using the above idiom. ;)

I'm not going to suggest yet another followup PR as if things are stable, lets just let it be. But the modern idiom in test_subprocess is to not launch things other than sys.executable based processes to avoid depending upon the external execution environment.